db/src/db.rs

```diff
@@ -457,17 +462,149 @@ pub enum SearchType {
     Inexact,
 }
 
-impl DB {
-
+/// `MentatStoring` will be the trait that encapsulates the storage layer.  It is consumed by the
+/// transaction processing layer.
+///
+/// Right now, the only implementation of `MentatStoring` is the SQLite-specific SQL schema.  In the
+/// future, we might consider other SQL engines (perhaps with different fulltext indexing), or
+/// entirely different data stores, say ones shaped like key-value stores.
+pub trait MentatStoring {
```

Comment:
There's conflicting precedent in Rust.

There are traits named after verbs: `Write`. After nouns: `Iterator`. After words that could be either: `Clone`.

Jim's book offers examples phrased as adjectives like `Spliceable` and `Visible` (with methods like `render`), named after their defining method`IsEmoji` (with method `is_emoji`), and named after a subtype (`Creature`).